### DISCUSSION TITLE
[Kavita] Suggestion for LNreader extension

### DISCUSSION THREAD

**Discussion by arkaeins** [Kareadita/Kavita#2886] [Ideas]
### Idea Description

It would be nice to have an extension for LNreader like Tachiyomi-Mihon.
Tachiyomi-Mihon only supports manga, so I can't read epubs that kavita supports. So if LNreader is supported by kavita, it will be possible to read epubs and manga with a third-party client.
Thank you for creating Kavita. This is a very useful thing.
I look forward to hearing from you.
(My English is not good, so I apologize if there are any mistakes).

### Idea Category

Feature Enhancement

### Duration of Using Kavita

2 weeks

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by scare376** [comment]
After looking over the documentation i would say its unlikely that it would be supported. The way that LNreader collects content to serve vs how kavita serves epubs is likely incompatible. https://github.com/LNReader/lnreader-plugins/blob/master/docs/docs.md from how i understand it kavita would essentially need to chop up the epub and serve the individual html pages to lnreader whereas it serves the whole epub file normally.

**Reply by scare376** [reply]
This is the same reason we cannot support image only manga epubs for mihon.

**Comment by bobthesponge10** [comment]
I recently made a basic proof of concept LNReader Kavita extension that seems to work for my use case.

It works by using the kavita api meant for the epub reader to get the page ranges for each chapter and then getting those pages one at a time and merging them together.

Github Repo: [https://github.com/bobthesponge10/lnreader-plugins](https://github.com/bobthesponge10/lnreader-plugins)
LNReader repo link: [https://raw.githubusercontent.com/bobthesponge10/lnreader-plugins/refs/heads/plugins/v3.0.0/.dist/plugins.min.json](https://raw.githubusercontent.com/bobthesponge10/lnreader-plugins/refs/heads/plugins/v3.0.0/.dist/plugins.min.json)

**Reply by majora2007** [reply]
Is this published somewhere? If it works, we can link it from the wiki.

**Reply by bobthesponge10** [reply]
I only have it published in my github at the moment. I'm planning on cleaning it up a bit and adding some features like filtering.

Eventually I would like it to be in the official LNReader plugins repo but until then it can be added with the above LNReader repo url.

**Reply by majora2007** [reply]
Awesome, I look forward to your final published version.

**Comment by djeinstine** [comment]
Hi @bobthesponge10 has there been any updates on this?

**Reply by bobthesponge10** [reply]
I haven't touched it in a while.

I wanted to wait for LNReader/lnreader#1459 to merge so I could implement syncing with the final version but it doesn't look like that's gonna happen for a while.

If there is demand I could finish it with its current features and implement syncing whenever that functionality gets added to LNReader.

For now the current repo I have linked above should still work.
